apps/base-docs/docs/building-with-base/guides/cross-chain-with-ccip.md

```diff
@@ -0,0 +1,615 @@
+    }
+}
+```
+
+Create a new file under your project’s `src/` directory named `Receiver.sol` and copy the code above into the file.
+
+#### Code walkthrough
+
+The sections below provide a detailed explanation for the code for the `Receiver` contract provided above.
+
+##### Initializing the contract
+
+In order to receive data using CCIP, the `Receiver` contract will need to extend to the`CCIPReceiver` interface. Extending this interface allows the `Receiver` contract to initialize the contract with the router address from the constructor, as seen below:
+
+```solidity
+import {CCIPReceiver} from "@chainlink/contracts-ccip/src/v0.8/ccip/applications/CCIPReceiver.sol";
+
+contract Receiver is CCIPReceiver {
+
+   /// @notice Constructor - Initializes the contract with the router address.
+   /// @param router The address of the router contract.
+   constructor(address router) CCIPReceiver(router) {}
+}
+```
+
+##### Receiving a message
+
+Extending the `CCIPReceiver` interface also allows the `Receiver` contract to override the `_ccipReceive` handler method for when a message is received and define custom logic.
+
+```solidity
+/// @notice Handle a received message
+/// @param message The cross-chain message being received.
+function _ccipReceive(
+    Client.Any2EVMMessage memory message
+) internal override {
+    // Add custom logic here
+}
+```
+
+The `Receiver` contract in this guide provides custom logic that stores the `messageId` and `text` (decoded) as member variables.
+
+```solidity
+contract Receiver is CCIPReceiver {
+
+   bytes32 private _messageId;
+   string private _text;
+
+   /// @notice Constructor - Initializes the contract with the router address.
+   /// @param router The address of the router contract.
+   constructor(address router) CCIPReceiver(router) {}
+
+   /// @notice Handle a received message
+   /// @param message The cross-chain message being received.
+   function _ccipReceive(
+       Client.Any2EVMMessage memory message
+   ) internal override {
+       _messageId = message.messageId; // Store the messageId
+       _text = abi.decode(message.data, (string)); // Decode and store the message text
+   }
+}
+```
+
+##### Retrieving a message
+
+The `Receiver` contract defines a custom method named `getMessage` that returns the details of the last received message `_messagId` and `_text`. This method can be called to fetch the message data details after the `_ccipReceive` receives a new message.
+
+```solidity
+/// @notice Gets the last received message.
+/// @return messageId The ID of the last received message.
+/// @return text The last received text.
+function getMessage()
+    external
+    view
+    returns (bytes32 messageId, string memory text)
+{
+    return (_messageId, _text);
+}
+```
+
+---
+
+## Compiling the Smart Contracts
+
+To compile your smart contracts, run:
+
+```bash
+forge build
+```
+
+---
+
+## Deploying the smart contract
+
+### Setting up your wallet as the deployer
```

Comment:
If this process isn't unique to cross-chain messaging, I'd link out to the other guide on how to deploy with Foundry.  Otherwise, we'll end up needing to maintain multiple articles when they inevitably change how deployment works.

Reply:
There's a trade-off here between code duplication and providing link outs for important steps within guides.

I would like the reader to not have to visit multiple different pages to find all of the steps if possible.

We can fix this by identifying places in our docs / guides that share duplicated content / steps, and make them into maintainable injectable templates that can be used across guides.

I lean towards minor content duplication > forcing reader to link out.

Reply:
I'll do a follow up PR that fixes this across the entire Base Docs site using - https://docusaurus.io/docs/markdown-features/react#importing-markdown